server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java

```diff
@@ -744,6 +744,18 @@ public int hashCode() {
         return globalCheckpoints;
     }
 
+    /**
+     * @return true iff any tracked global checkpoint for an in-sync copy lags behind our global checkpoint
+     */
+    public synchronized boolean trackedGlobalCheckpointsNeedSync() {
```

Comment:
Why do we need to synchronize here when previously we didn't synchronize while running the same loop?
Was this a bug before?

Reply:
Previously we were `synchronised` in the loop within `ReplicationTracker#getInSyncGlobalCheckpoints()` that iterates over `ReplicationTracker#checkpoints`.